integrationtests/wildfly-modules/src/test/java/org/infinispan/test/integration/as/query/ElasticQueryConfiguration.java

```diff
@@ -0,0 +1,31 @@
+package org.infinispan.test.integration.as.query;
+
+import java.io.IOException;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Disposes;
+import javax.enterprise.inject.Produces;
+
+import org.infinispan.Cache;
+import org.infinispan.manager.DefaultCacheManager;
+import org.infinispan.manager.EmbeddedCacheManager;
+
+/**
+ * @since 9.0
+ */
+@ApplicationScoped
+public class ElasticQueryConfiguration {
+
+   @Produces
+   @ApplicationScoped
+   public Cache<String, Book> defaultClusteredCacheManager() throws IOException {
+      return new DefaultCacheManager("elasticsearch-indexing.xml").getCache();
+   }
+
+   public void killCacheManager(@Disposes Cache<String, Book> cache) {
```

Comment:
You could use CDI integration here (just add a dependency in MANIFEST.MF to `org.infinispan.cdi.embedded`). This way the cleanup will be done for you and all you need to do is to create an `EmbeddedCacheManager`.